library/internal/commands.go

```diff
@@ -0,0 +1,94 @@
+package internal
+
+import (
+	"io"
+	"os/exec"
+	"syscall"
+)
+
+// StartedCommand is an executor.Command that has successfully started
+type StartedCommand struct {
+	Cmd          *exec.Cmd
+	CmdStatus    string
+	OutputBuffer *MultiplexingBuffer
+	AfterDone    func()
+}
+
+func (c *StartedCommand) Output() io.Reader {
+
+	return c.OutputBuffer.Reader()
+}
+
+// Stop will send a SIGTERM signal to the underlying process
+func (c *StartedCommand) Stop() error {
+	err := c.Cmd.Process.Signal(syscall.SIGTERM)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ForceStop will send a SIGKILL signal to the underlying process
+func (c *StartedCommand) ForceStop() error {
+	err := c.Cmd.Process.Kill()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *StartedCommand) Status() string {
+	return c.CmdStatus
+}
+
+//WaitForStatus will wait until the underlying process complentes, then executes cleanup and sets the status
+func (c *StartedCommand) WaitForStatus() {
+
+	err := c.Cmd.Wait()
+
+	_ = c.OutputBuffer.Close()
+	c.AfterDone()
+
+	if err != nil {
+		c.CmdStatus = ABORTED
```

Comment:
What if multiple clients are reading the command status simultaneously with it being updated?